Automattic/jetpack
# Search: change connectScreen so it is a component instead of the renderConnectScreen component

Change connectScreen to be a component to fix unnecessary re-renders and lack of visibility in React Dev tools.



#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No
#### Testing instructions:
1- Launch a JN site with Search standalone and this branch
2- Keep the JS console open and expect to see no warning for the following steps
3- Expect to be able to Connect and navigate the page.
4- Expect to see no crashing page
